File: 0455-assign-cookies/0455-assign-cookies.cpp
class Solution {
public:
    int findContentChildren(vector<int>& g, vector<int>& s) {
        int m=s.size(),n=g.size(),l=0,r=0;
        sort(g.begin(),g.end());
        sort(s.begin(),s.end());
        while(l<m && r<n){
            if(g[r]<=s[l]){
                r++;
            }
            l++;
        }
        return r;
    }
};

File: 0455-assign-cookies/0455-assign-cookies.py
class Solution:
    def findContentChildren(self, g: List[int], s: List[int]) -> int:
        child_i,cookie_i=0,0
        s.sort()
        g.sort()
        while child_i<len(g) and cookie_i<len(s):
            if s[cookie_i]>=g[child_i]:
                child_i+=1
            cookie_i+=1
        return child_i
